dockertest/subtestbase.py

```diff
@@ -155,8 +157,15 @@ def failif_not_in(needle, haystack, description=None):
             description = 'string'
         if needle in haystack:
             return
-        raise DockerTestFail("Expected string '%s' not in %s '%s'"
-                             % (needle, description, haystack))
+        expect_s = "Expected string '%s'" % needle
+        if '|' in needle:
+            needles = [n.strip() for n in needle.split('|')]
+            expect_s = "Expected strings %s" % needles
+            for subneedle in needles:
+                if subneedle.strip() in haystack:
+                    return
+        raise DockerTestFail("%s not in %s '%s'" %
+                             (expect_s, description, haystack))
```

Comment:
Can we say with confidence that the '|' character will never be needed as part of the search string?  If not, then an escaping-mechanism will be needed, along with escapes for the escape character as well.  An easier implementation maybe to add a ``delimiter="|"`` parameter, which will work with existing code, yet allow the multiple-strings feature as well.  It would also permit disabling the feature with e.g. ``delimiter = None``.